OpenRA.Mods.Common/Traits/BotModules/HarvesterBotModule.cs

```diff
@@ -273,5 +432,25 @@ void INotifyActorDisposing.Disposing(Actor self)
 			if (resourceLayer != null)
 				resourceLayer.CellChanged -= ResourceCellChanged;
 		}
+
+		void IBotRespondToAttack.RespondToAttack(IBot bot, Actor self, AttackInfo e)
+		{
+			if (respondToAttackCooldown > 0 || !Info.HarvesterTypes.Contains(self.Info.Name)
+				|| e.Attacker == null || e.Attacker.IsDead || !e.Attacker.AppearsHostileTo(self))
+				return;
+
+			var parach = self.TraitOrDefault<Parachutable>();
+			if (parach != null && parach.IsInAir)
+				return;
+
+			var dockClientManager = self.Trait<DockClientManager>();
+			if (dockClientManager.ReservedHostActor != null)
+				return;
+
+			respondToAttackCooldown = 30;
+			var scanFromActor = dockClientManager.ClosestDock(null, ignoreOccupancy: true)?.Actor;
```

Comment:
```suggestion
			var scanFromActor = dockClientManager.ClosestDock(null, forceEnter: true, ignoreOccupancy: true)?.Actor;
```

We should also force enter, to make sure that the harvester goes even if it's empty